### PR TITLE
Remove getThisTimeWindowRemainder(), use getNextTimestepMaxLength() instead 

### PR DESCRIPTION
**Pull request by BenjaminRodenberg**
## Main changes of this PR

Removes function `getThisTimeWindowRemainder()` and uses `getNextTimestepMaxLength()` instead, because both functions have the same functionality.

## Motivation and additional information

Simplification of code base.

## Author's checklist

* [x] I used the [`pre-commit` hook](https://precice.org/dev-docs-dev-tooling.html#setting-up-pre-commit) to prevent dirty commits and used `pre-commit run --all` to format old commits.
* [x] ~~I added a changelog file with `make changelog` if there are user-observable changes since the last release.~~ not necessary
* [x] ~~I added a test to cover the proposed changes in our test suite.~~ pure refactoring, but added a few tests in this direction in [3e82443](https://github.com/precice/precice/commit/3e824439a4e4ac477b5f5aafe1976162f3105143)
* [x] I sticked to C++17 features.
* [x] I sticked to CMake version 3.16.3.
* [x] I squashed / am about to squash all commits that should be seen as one.

## Reviewers' checklist

* [ ] Does the changelog entry make sense? Is it formatted correctly?
* [x] Do you understand the code changes?

### CODE REVIEW

**Comment by BenjaminRodenberg** [comment]
@uekerman Do you have an idea where we should additionally do testing here?

File: src/cplscheme/BaseCouplingScheme.cpp [hunk 1]
```diff
@@ -687,7 +676,7 @@ void BaseCouplingScheme::advanceTXTWriters()
 
 bool BaseCouplingScheme::reachedEndOfTimeWindow()
 {
-  return math::equals(getThisTimeWindowRemainder(), 0.0, _eps);
+  return math::equals(getNextTimestepMaxLength(), 0.0, _eps) || not hasTimeWindowSize();
```
Comment (BenjaminRodenberg) [inline src/cplscheme/BaseCouplingScheme.cpp:679]: I think this is the most critical change with respect to functionality of `getNextTimestepMaxLength` vs. `getThisTimeWindowRemainder`.

File: src/cplscheme/BaseCouplingScheme.hpp [hunk 2]
```diff
@@ -140,21 +140,13 @@ class BaseCouplingScheme : public CouplingScheme {
    */
   double getTimeWindowSize() const override final;
 
-  /**
-   * @brief Returns the remaining timestep length within the current time window.
-   *
-   * If no time window size is prescribed by the coupling scheme, always 0.0 is
```
Comment (BenjaminRodenberg) [inline src/cplscheme/BaseCouplingScheme.hpp:146]: Returning 0.0 here I think is not a good idea (looking at the behavior of `method="participant-first"`

**Review by uekerman — APPROVED** [review]
Makes sense :+1:

**Pull request merged**